io/GDALWriter.cpp

```diff
@@ -321,6 +329,45 @@ void GDALWriter::doneFile()
         throwError(raster.errorMsg());
 
     getMetadata().addList("filename", m_filename);
+
+    std::vector<std::string> gdalitems = Utils::split(m_GDAL_metadata, ',');
+    for (auto& v: gdalitems)
+    {
+        const std::size_t pos = v.find_first_of("=");
+        if (pos != std::string::npos)
+        {
+            const std::string name = v.substr(0, pos);
+            const std::string value = pos != std::string::npos ? v.substr(pos + 1) : "";
+            raster.addMetadata(name, value);
+        }
+    }
+
+    getMetadata().addList(raster.getMetadata());
```

Comment:
I don't think you want arrays as output. The only reason `addList` exists is to create an array of one item.

Reply:
a GDAL raster might have multiple metadata entries. This puts them all in the `raster` metadata entry as an array of key/value items, which is what we want.

Reply:
I don't care, but isn't that just an object?

```
#include <iostream>

#include <pdal/Metadata.hpp>
#include <pdal/PDALUtils.hpp>

int main()
{
    using namespace pdal;

    MetadataNode raster("raster");

    raster.add("foo", "fooval");
    raster.add("bar", "barval");

    MetadataNode root("readers.gdal");
    root.add(raster);

    Utils::toJSON(root, std::cerr);

    return 0;
}
```
```
{
  "readers.gdal":
  {
    "raster":
    {
      "bar": "barval",
      "foo": "fooval"
    }
  }
}
```

Reply:
updated as suggested.